platforms/esp/32/fastpin_esp32.h

```diff
@@ -24,68 +11,90 @@ template<uint8_t PIN, uint32_t MASK> class _ESPPIN {
   inline static void setOutput() { pinMode(PIN, OUTPUT); }
   inline static void setInput() { pinMode(PIN, INPUT); }
 
-  inline static void hi() __attribute__ ((always_inline)) { if(PIN < OUTPUT_PIN_LIMIT) { _GPB0._GPOS = MASK; } }
-  // inline static void hi() __attribute__ ((always_inline)) { gpio_set_level((gpio_num_t)PIN, HIGH); }
+  inline static void hi() __attribute__ ((always_inline)) { 
+      if (PIN < 32) GPIO.out_w1ts = MASK;
+      else if(PIN < 34) GPIO.out1_w1ts.val = MASK;
```

Comment:
Can you make this >= 32?  This way if an esp32 board comes out that exposes more pins on the secondary controller, this will still work.  Ditto for the other set below

Reply:
No problem. I copied that log directly from the ESP-IDF core, though. I think the reason they write it like that is that there are pins 35-39, but they are input only.

Reply:
Espressif's internal headers seem to indicate that 35-39 could also be for out - from https://github.com/espressif/esp-idf/blob/master/components/soc/esp32/include/soc/gpio_struct.h -- 

```
    union {
        struct {
            uint32_t data:       8;                 /*GPIO32~39 output value*/
            uint32_t reserved8: 24;
        };
        uint32_t val;
    } out1;
    union {
        struct {
            uint32_t data:       8;                 /*GPIO32~39 output value write 1 to set*/
            uint32_t reserved8: 24;
        };
        uint32_t val;
    } out1_w1ts;
    union {
        struct {
            uint32_t data:       8;                 /*GPIO32~39 output value write 1 to clear*/
            uint32_t reserved8: 24;
        };
        uint32_t val;
    } out1_w1tc;
```

Reply:
Totally up to you. I'm using the Espressif implementations of the various pin functions as my guide (e.g., their version of digitalWrite) -- it seemed like the most likely to work. I can try something different, but I'll have to do some testing to make sure it works.